code/modules/vehicles/bike.dm

```diff
@@ -105,16 +145,33 @@
 		return
 
 /obj/vehicle/bike/attack_hand(var/mob/user as mob)
-	if(user == load)
-		unload(load)
-		to_chat(user, "You unbuckle yourself from \the [src]")
+	if(key)
+		to_chat(user, "You take \the [key] out of \the [src]")
+		user.put_in_hands(key)
+		key = null
+		if(on)
+			toggle_engine(user)
 	else if(user != load && load)
 		user.visible_message ("[user] starts to unbuckle [load] from \the [src]!")
 		if(do_after(user, 8 SECONDS, src))
 			unload(load)
 			to_chat(user, "You unbuckle [load] from \the [src]")
 			to_chat(load, "You were unbuckled from \the [src] by [user]")
 
+/obj/vehicle/bike/attackby(obj/item/item, mob/living/user)
```

Comment:
Preserve the header like every other attackby proc

Reply:
done, if you mean to the item var should be `attacking_item`

Reply:
not only, the second parameter is a mob/user, not a mob/living/user

Reply:
done